File: backend/app/schemas/conversation.py
# ============================================================
# 第二分身 schema（对齐前端 types/conversation.ts）
# ============================================================
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ConversationOut(BaseModel):
    id: str
    title: str | None = None
    scene: str = "general"
    created_at: datetime
    updated_at: datetime


class CreateConversationRequest(BaseModel):
    title: str | None = None


class MessageOut(BaseModel):
    id: str
    conversation_id: str
    role: str
    content: str
    tokens: int | None = None
    referenced_doc_ids: list[str] = []
    created_at: datetime


class SendMessageRequest(BaseModel):
    content: str = Field(min_length=1)
    referenced_doc_ids: list[str] = []
    mode: str = "normal"  # normal | deep | creative | critical | brainstorm


class SseEvent(BaseModel):
    type: str  # content | done | error
    content: str | None = None
    tokens: int | None = None
    error: str | None = None


class AiSummaryResult(BaseModel):
    summary: str


class AiTagsResult(BaseModel):
    tags: list[str]


class AiInspirationResult(BaseModel):
    inspiration: str
    direction: list[str] = []
    prompt: str = ""


class ReflectionQuestion(BaseModel):
    question: str
    answer: str | None = None


class SummarizeRequest(BaseModel):
    document_id: str


class SuggestTagsRequest(BaseModel):
    document_id: str


class InspirationRequest(BaseModel):
    document_id: str


class ReflectionQuestionsRequest(BaseModel):
    review_type: str = "daily"
    date: str = ""
    data: dict = {}

File: frontend/src/types/conversation.ts
// ============================================================
// 第二分身（AI对话）模块类型
// 对应 PRD §13.5 第二分身模块 + §15.8-15.9 表结构
// ============================================================

export type MessageRole = 'user' | 'assistant' | 'system'

export interface Conversation {
  id: string
  title: string | null
  scene: string
  created_at: string
  updated_at: string
}

export interface Message {
  id: string
  conversation_id: string
  role: MessageRole
  content: string
  tokens: number | null
  referenced_doc_ids: string[]
  created_at: string
}

export interface SendMessageRequest {
  content: string
  referenced_doc_ids?: string[]
  /** 思维模式：normal | deep | creative | critical | brainstorm */
  mode?: string
}

/** SSE 流式响应事件 */
export type SseEventType = 'content' | 'done' | 'error'

export interface SseEvent {
  type: SseEventType
  content?: string
  tokens?: number
  error?: string
}

export interface AiSummaryResult {
  summary: string
}

export interface AiTagsResult {
  tags: string[]
}

export interface AiInspirationResult {
  inspiration: string
  direction: string[]
  prompt: string
}

export interface ReflectionQuestion {
  question: string
  answer?: string
}
